### PR TITLE
Enable method chaining EventDispatcher

### PR DESCRIPTION
**Pull request by takahirox**
I think it'd be useful if `EventDispatcher`'s methods support method chaining. This PR enables it.

```javascript
const object = new Foo()
    .addEventListener('foo', (event) => { ... })
    .addEventListener('bar', (event) => { ... });
```

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
Oh, Travis is failing because of this message:

> src/core/DirectGeometry.d.ts:47:2 - error TS2416: Property 'addEventListener' in type 'DirectGeometry' is not assignable to the same property in base type 'EventDispatcher'.
  Type '(type: string, listener: (event: Event) => void) => void' is not assignable to type '(type: string, listener: (event: Event) => void) => EventDispatcher'.
    Type 'void' is not assignable to type 'EventDispatcher'.
47  addEventListener( type: string, listener: ( event: Event ) => void ): void;

It seems `DirectGeometry.d.ts` and `Geometry.d.ts` redefine the event dispatcher API for some reasons. I think those can be removed.

**Comment by takahirox** [comment]
Oops, I should've checked that. I'll fix. Thanks.

File: src/core/DirectGeometry.js [hunk 1]
```diff
@@ -34,7 +35,7 @@ function DirectGeometry() {
 
 }
 
-Object.assign( DirectGeometry.prototype, {
+DirectGeometry.prototype = Object.assign( Object.create( EventDispatcher.prototype ), {
```
Comment (takahirox) [inline src/core/DirectGeometry.js]: It seems missing EventDispatcher inherit. I added.

**Comment by takahirox** [comment]
Updated

**Comment by Mugen87** [comment]
Looks good!

File: src/core/DirectGeometry.js (hunk 1 above)
Comment (Mugen87) [reply src/core/DirectGeometry.js]: Um, I don't think this inheritance is needed. No `EventDispatcher` method is used.

Comment (takahirox) [reply src/core/DirectGeometry.js]: So removing the inheritance from JS and TS?

Comment (takahirox) [reply src/core/DirectGeometry.js]: Removed

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
I did [EventDispatcher.js](https://github.com/mrdoob/eventdispatcher.js) because Javascript native event system didn't support extending it. I was hoping to replace that code with JavaScript's own implementation some day so I tried to copy the `EventTarget` API as best as I could.

Unfortunately, `EventTarget.addEventListener` returns `undefined` and I don't think deviating from that is a good idea.

**Comment by Mugen87** [comment]
Oh, I see. I was not aware that you want to keep the API of  `EventDispachter` close to `EventTarget`^^.

Even if the PR is not merged, the changes to the TS declaration files of `Geometry` and `DirectGeometry` still make sense. However, better to do this in a separate PR then.

**Comment by takahirox** [comment]
OK, I can understand. Some notes. (I wish `EventTarget` methods would return `this`...)

In ES6 `EventTarget` can be extended. When we get rid of ES5 support at some point, we may switch to `EventTarget`.

```javascript
class Foo extends EventTarget {
  constructor() {
    super();
  }
}

const foo = new Foo();
foo.addEventListener('var', event => console.log(event));
foo.dispatchEvent(new Event('var'));
-> Event { ... }
```

And I made TS cleanup PR #17440. I close this PR.

**Pull request closed without merging**